File: Irrgarten_Ruby/Irrgarten_Ruby_P4/CombatElement.rb
#encoding:utf-8

module Irrgarten 
	class CombatElement
		
		def initialize(effect,uses) #float,int
			@effect=effect
			@uses=uses
		end
		
		
		
		def discard()
			return Dice.discard_element(@uses)
		end
		
		def to_s()
			"[#{@effect},#{@uses}]"
		end
		
		protected
		
		def produce_effect()
			salida = 0.0
			if @uses > 0 then 
				salida = @effect
				@uses -= 1
			end
			salida
		end
		
		def to_s()
			salida = "[#{@effect},#{@uses}]"
			salida
		end
	end
end

File: Irrgarten_Java/Irrgarten_P5/src/irrgarten/CombatElement.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author leonin04
 */
abstract public class CombatElement {
    private float effect;
    private int uses;
    
    public CombatElement( float effect, int uses){
        this.effect = effect;
        this.uses = uses;
    }
    
    protected float produceEffect(){
        float salida = 0.0f;
        
        if (uses > 0) {
            salida = effect;
            uses--;
        }
        return salida;
    }
    
    public boolean discard(){
        return Dice.discardElement(this.uses);
    }
    
    @Override
    public String toString(){
        return "[" + effect + "," + uses + "]";
    }
}
